api/src/main/java/org/openmrs/validator/VisitValidator.java

```diff
@@ -159,6 +165,44 @@ private void validateStopDatetime(Visit visit, Visit otherVisit, Errors errors)
 			errors.rejectValue("stopDatetime", "Visit.visitCannotContainAnotherVisitOfTheSamePatient", messageBuilder
 			        .toString());
 		}
-		
+
+	}
+
+	private void validateStartDateTimeToBirthdate(Visit visit, Errors errors) {
+
+		Patient patient = visit.getPatient();
+
+		if (patient != null) {
+			if (visit.getStartDatetime() != null && patient.getBirthdate() != null) {
+				if (!patient.getBirthdateEstimated() && visit.getStartDatetime().before(patient.getBirthdate())) {
+					errors.rejectValue("startDatetime", "Visit.startDateCannotBeBeforeBirthdate",
+							"This visit has a start date before birth date of the patient.");
+				}
+
+				if (patient.getBirthdateEstimated()) {
+
+					Calendar calendar = Calendar.getInstance();
+					calendar.setTime(patient.getBirthdate());
+
```

Comment:
I would say for estimated birthdate and in general birthdate precision when comparing it to other dates should be to day and ignoring the time